### ISSUE TITLE
Describe parameter dev options used for research 

### ISSUE THREAD

**Issue by k-doering-NOAA** [nmfs-stock-synthesis/stock-synthesis#343]
I found this old [draft documentation](https://docs.google.com/document/d/1ZUy4gw9WITmSFdqD2jhk-EvWb4TVPzzkIXSsV_rKFww/edit?usp=sharing) regarding parameter deviations that made me realize I should double check the available options for annual deviation link options.

Following the [Long parameter line section of the manual](https://nmfs-stock-synthesis.github.io/doc/SS330_User_Manual.html#specification-of-time-varying-parameters-long-parameter-lines), only options 1-6 and 21-26 are available, which I believe makes sense (the old document suggests that 11-16 and 31-36 also exist, but that they are incorrect, so it makes sense that they not be used or documented).

However, I did find that I think there is still code for using at least the 11-16 options, e.g., :https://github.com/nmfs-stock-synthesis/stock-synthesis/blob/635c10f0b789d41538087b3533d9672a7df5cc75/SS_readcontrol_330.tpl#L3972-L3975

I think this code should be removed, and SS3 should exit on error if a user tries to use options 11-16 or 31-36.

**Comment by Rick-Methot-NOAA** [comment]
duly added to my growing list.

**Comment by Rick-Methot-NOAA** [comment]
internally, the dev type is stored in timevary_setup(8) after being read as the DevLink from a parameter line
#_DevLinks:  1: P(y)*=exp(dev(y)*dev_se;  2: P(y)+=dev(y)*dev_se;  3: random walk;  4: zero-reverting random walk with rho;  5: like 4 with logit transform to stay in base min-max
#_DevLinks(more):  21-25 keep last dev for rest of years

In addition, because 2DAR also uses parmdevs, there is a flag to differentiate them using parm_dev_type with values from 1 to 4.
parmdev_type = 1 is used for parameter devs
parmdev_type = 2 is used internally to indicate 2DAR

In addition 2 other codes are used internally in SS_objfun.tpl for testing only.  They are not documented, which is OK.
parmdev_type = 3 is assigned if the DevLink is >10 and <20.  This invokes:
use objfun using -log(se) to match 3.30.12 and earlier
parmdev_type = 4 is assigned if the DevLink is 6.  This adds a penalty to the objfun to keep the rmse near 1.0.   It is used in SS_timevaryparm().  This approach works, but slows final convergence because getting stddev exactly to 1.0 causes high correlation among devs.

In conclusion, I will keep these options in the code for testing as the parameter deviation approach is still under development.
